serf/event.go

```diff
@@ -156,13 +158,53 @@ func (q *Query) Respond(buf []byte) error {
 		return fmt.Errorf("response exceeds limit of %d bytes", q.serf.config.QueryResponseSizeLimit)
 	}
 
-	// Send the response
-	addr := net.UDPAddr{IP: q.addr, Port: int(q.port)}
-	if err := q.serf.memberlist.SendTo(&addr, raw); err != nil {
-		return err
+	// Relay the response through up to QueryResponseRelayLimit nodes
+	members := q.serf.Members()
+	relayMembers := kRandomMembers(q.serf.config.QueryResponseRelayLimit, members, func(m Member) bool {
```

Comment:
I wonder if the relay factor should come from a field in the query itself? That way the requestor can set the importance of it, and queries from an originator that doesn't support dedup will naturally have this set to 0, which is kind of nice.